src/ddb/ddb_main.c

```diff
@@ -59,12 +72,22 @@ run_cmd(struct ddb_ctx *ctx, struct argv_parsed *parse_args)
 		rc = ddb_run_dump_value(ctx, &info.dci_cmd_option.dci_dump_value);
 		break;
 	case DDB_CMD_RM:
+		if (!write_mode) {
+			ddb_print(ctx, error_msg_write_mode_only);
+			rc = -DER_INVAL;
+			break;
+		}
 		rc = ddb_run_rm(ctx, &info.dci_cmd_option.dci_rm);
 		break;
 	case DDB_CMD_DUMP_DTX:
 		rc = ddb_run_dump_dtx(ctx, &info.dci_cmd_option.dci_dump_dtx);
 		break;
 	case DDB_CMD_LOAD:
+		if (!write_mode) {
```

Comment:
The name "LOAD" is some misguided: why load need "write" permission?

Reply:
LOAD modifies the VOS tree. It 'loads' a new value into an akey if it exists, or creates a new akey. Thoughts on a better name?

Reply:
Either more clear name (although I do not have a good suggested name...) or some comment here to avoid misguiding.